solomonhawk/src/day1.rs

```diff
@@ -1,31 +1,58 @@
 #[aoc_generator(day1)]
 pub fn input_generator(input: &str) -> Vec<usize> {
-    input
-        .trim()
-        .split("\n")
-        .map(|n| n.parse().unwrap())
-        .collect()
+    input.split("\n").map(|n| n.parse().unwrap()).collect()
 }
 
 #[aoc(day1, part1)]
 pub fn part1(nums: &[usize]) -> usize {
-    for n in nums.iter() {
-        for m in nums.iter() {
-            if n + m == 2020 {
-                return n * m;
-            }
-        }
-    }
+    count_increasing_seqs(nums)
+}
+
+#[aoc(day1, part2)]
+pub fn part2(nums: &[usize]) -> usize {
+    count_increasing_seqs(
+        &nums
+            .windows(3)
+            .map(|ns| ns.iter().sum())
+            .collect::<Vec<usize>>(),
+    )
+}
 
-    unreachable!()
+pub fn count_increasing_seqs(nums: &[usize]) -> usize {
+    nums.windows(2)
+        .fold(0, |acc, slice| acc + (slice[1] > slice[0]) as usize)
```

Comment:
I like doing it one step here.  I ended up doing a length of a filtered list.  I think they are pretty much equivalent since Haskell is lazy.  Also clever casting the boolean to a number.